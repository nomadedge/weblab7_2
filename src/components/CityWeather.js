import React, { Component } from 'react';

import '../styles/Weather.css';

class CityWeather extends Component {
    render() {
        const { cityWeather } = this.props;
        return (
            <>
                <div id='left-block'>
                    <div id='city-name'>{cityWeather.name}</div>
                    <div id='icon-temp'>
                        <img src={cityWeather.weather.iconUrl} alt='Weather icon' />
                        <div id='temp'>{cityWeather.weather.temperature}Â°C</div>
                    </div>
                </div>
                <div id='right-block'>
                    <div className='info'>
                        <div>Description:</div>
                        <div className='value'>{cityWeather.weather.description}</div>
                    </div>
                    <div className='info'>
                        <div>Barometer:</div>
                        <div className='value'>{cityWeather.weather.pressure} hPa</div>
                    </div>
                    <div className='info'>
                        <div>Humidity:</div>
                        <div className='value'>{cityWeather.weather.humidity} %</div>
                    </div>
                    <div className='info'>
                        <div>Wind:</div>
                        <div className='value'>{cityWeather.weather.windSpeed} m/s</div>
                    </div>
                </div>
            </>
        );
    }
}

export default CityWeather;