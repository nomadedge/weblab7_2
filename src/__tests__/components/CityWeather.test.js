import React from 'react';
import CityWeather from '../../components/CityWeather';
import renderer from 'react-test-renderer';

describe('CityWeather component', () => {
    test('matches the snapshot', () => {
        const cityWeather = {
            name: 'correctCityName',
            isFetching: false,
            error: null,
            weather: {
                iconUrl: "http://openweathermap.org/img/w/50n.png",
                description: "mist",
                temperature: "14",
                pressure: 1022,
                humidity: 100,
                windSpeed: 3.6
            }
        };

        const tree = renderer.create(
            <CityWeather cityWeather={cityWeather} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});