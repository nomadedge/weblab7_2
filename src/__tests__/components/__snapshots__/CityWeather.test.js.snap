// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityWeather component matches the snapshot 1`] = `
Array [
  <div
    id="left-block"
  >
    <div
      id="city-name"
    >
      correctCityName
    </div>
    <div
      id="icon-temp"
    >
      <img
        alt="Weather icon"
        src="http://openweathermap.org/img/w/50n.png"
      />
      <div
        id="temp"
      >
        14
        Â°C
      </div>
    </div>
  </div>,
  <div
    id="right-block"
  >
    <div
      className="info"
    >
      <div>
        Description:
      </div>
      <div
        className="value"
      >
        mist
      </div>
    </div>
    <div
      className="info"
    >
      <div>
        Barometer:
      </div>
      <div
        className="value"
      >
        1022
         hPa
      </div>
    </div>
    <div
      className="info"
    >
      <div>
        Humidity:
      </div>
      <div
        className="value"
      >
        100
         %
      </div>
    </div>
    <div
      className="info"
    >
      <div>
        Wind:
      </div>
      <div
        className="value"
      >
        3.6
         m/s
      </div>
    </div>
  </div>,
]
`;
